src/Core/Kpi/Row/KpiRowFactory.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Kpi\Row;
+
+use PrestaShop\PrestaShop\Core\Kpi\KpiInterface;
+
+/**
+ * Class KpiRowFactory builds a KPI row
+ */
+final class KpiRowFactory implements KpiRowFactoryInterface
+{
+    /**
+     * @var KpiInterface[]
+     */
+    private $kpis;
+
+    /**
+     * @param KpiInterface ...$kpis
+     */
+    public function __construct(KpiInterface ...$kpis)
```

Comment:
this won't works in PHP 5.6, isn't it? I'm sad because it's an elegant implementation :/

Reply:
I think it will: http://php.net/manual/en/functions.arguments.php#functions.variable-arg-list

Reply:
great!